examples/self_supervised_learning/losses/_wav2vec2_loss.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional, Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+
+
+def compute_contrastive_loss(x, mask_indices, targets, neg_is_pos, reduce, logit_temp = 0.1):
+    x = (
+        x[mask_indices]
+        .view(x.size(0), -1, x.size(-1))
+        .unsqueeze(0)
+        .expand(targets.shape)
+    )
+    logits = torch.cosine_similarity(x.float(), targets.float(), dim=-1).float()
+    logits /= logit_temp
+    if neg_is_pos.any():
+        logits[1:][neg_is_pos] = float("-inf")
+    target = logits.new_zeros(logits.size(1) * logits.size(2), dtype=torch.long)
+    logits = logits.transpose(0, 2)
+    logits = logits.reshape(-1, logits.size(-1))
+    loss = F.cross_entropy(
+        logits,
+        target,
+        reduction="sum" if reduce else "none",
+    )
+    sample_size = target.numel()
+    return loss, sample_size, logits
```

Comment:
I think we can remove the logits from output, since it is not used in the loss function
```suggestion
    return loss, sample_size
```